rsconnect/bundle.py

```diff
@@ -747,6 +748,69 @@ def make_html_bundle(
     return bundle_file
 
 
+def make_voila_bundle(
+    path: str,
+    entrypoint: str,
+    extra_files: typing.List[str],
+    excludes: typing.List[str],
+    force_generate: bool,
+    environment: Environment,
+    image: str = None,
+) -> typing.IO[bytes]:
+    """
+    Create an voila bundle, given a path and/or entrypoint.
+
+    The bundle contains a manifest.json file created for the given notebook entrypoint file.
+    If the related environment file (requirements.txt) doesn't
+    exist (or force_generate is set to True), the environment file will also be written.
+
+    :param path: the file, or the directory containing the files to deploy.
+    :param entry_point: the main entry point.
+    :param extra_files: a sequence of any extra files to include in the bundle.
+    :param excludes: a sequence of glob patterns that will exclude matched files.
+    :param force_generate: bool indicating whether to force generate manifest and related environment files.
+    :param image: the optional docker image to be specified for off-host execution. Default = None.
+    :return: a file-like object containing the bundle tarball.
+    """
+    mimetypes.add_type("text/ipynb", ".ipynb")
```

Comment:
I'm not sure it matters for how we are using this, but the mimetype for ipynb files appears to be [application/x-ipynb+json](https://github.com/jupyter/notebook/pull/6010/files#diff-b34aa95c7163adbb4c323d5930b11fbbc28ff11984f6436fb139358f7ba5b719)